packages/ag-charts-enterprise/src/components/dialog/dialog.ts

```diff
@@ -28,29 +30,57 @@ interface TextAreaOptions {
     onChange?: (value: string) => void;
 }
 
+interface ColorPickerOptions {
+    altText: string;
+    label: string;
+    value: string | undefined;
+    onChange: (colorOpacity: string, color: string, opacity: number) => void;
+}
+
 /**
  * A popover that opens at the bottom right of the chart but can be moved by the user.
  */
 export abstract class Dialog<Options extends DialogOptions = DialogOptions> extends Popover<Options> {
-    private static readonly offset = 100;
+    private static readonly offset = 60;
 
+    private readonly colorPicker = new ColorPicker(this.ctx);
+    private colorPickerAnchorElement?: HTMLElement;
     private dragStartState?: { client: _Util.Vec2; position: _Util.Vec2 };
+    private seriesRect?: _Scene.BBox;
+
+    constructor(ctx: _ModuleSupport.ModuleContext, id: string) {
+        super(ctx, id);
+        this.destroyFns.push(
+            ctx.layoutManager.addListener('layout:complete', this.onLayoutComplete.bind(this)),
+            ctx.interactionManager.addListener('keydown', this.onKeyDown.bind(this))
+        );
+    }
 
     protected override showWithChildren(children: Array<HTMLElement>, options: Options) {
         const popover = super.showWithChildren(children, options);
         popover.classList.add('ag-charts-dialog');
         popover.setAttribute('role', 'dialog');
 
-        // Give the dialog's dimensions a chance to be calculated before positioning
-        requestAnimationFrame(() => {
-            const bbox = this.ctx.domManager.getBoundingClientRect();
-            const position = Vec2.sub(Vec2.sub(Vec2.from(bbox), Vec2.from(popover)), Dialog.offset);
-            this.updatePosition(position);
+        popover.addEventListener('mousedown', (event) => {
```

Comment:
The popover should probably stop propagation of events so you don't need to detect this here

Reply:
Or is this just a button click?

Reply:
Yeh, the color picker button triggers opening the color picker, and I want to hide it again if you click anywhere within the dialog. Needed to block it from immediately hiding the color picker when you click the button the first time, but thinking about it, there may be a better approach.